File: templates/matlab/stream_graph.m
function fig = stream_graph()
    addpath(fullfile(fileparts(mfilename('fullpath')),'..','..','_utils','matlab'));
    apply_theme(); rng(3);
    n = 200; n_series = 5;
    x = linspace(0, 24, n);
    Y = zeros(n_series, n);
    for i = 1:n_series
        for b = 1:4                  % each stream = sum of gaussian bumps
            c = 2 + 20*rand; wdt = 1.5 + 3.5*rand; a = 0.5 + 1.5*rand;
            Y(i, :) = Y(i, :) + a * exp(-0.5 * ((x - c) / wdt).^2);
        end
    end
    base = -sum(Y, 1) / 2;           % wiggle baseline, symmetric silhouette
    layers = [zeros(1, n); cumsum(Y, 1)] + base;
    fig = figure('Position',[100 100 800 400]); hold on;
    hs = gobjects(1, n_series); labels = cell(1, n_series);
    for i = 1:n_series
        hs(i) = fill([x fliplr(x)], [layers(i,:) fliplr(layers(i+1,:))], ...
                     palette('cat',i), 'FaceAlpha', 0.85, ...
                     'EdgeColor', [1 1 1], 'LineWidth', 0.5);
        labels{i} = sprintf('topic %d', i);
    end
    xlabel('time (month)'); ylabel('flow magnitude');
    title('Stream graph');
    legend(hs, labels, 'Location', 'southoutside', 'Orientation', ...
           'horizontal', 'Box', 'off');
    set(gca, 'XGrid', 'on', 'YGrid', 'off');
end

File: templates/python/stream_graph.py
"""stream_graph: 河流图（堆叠面积 wiggle 基线，ThemeRiver 风格）."""
import sys, os; sys.path.insert(0, os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', '..', '_utils', 'python'))
import numpy as np
import matplotlib.pyplot as plt
from theme import apply_theme
from palette import cycle

def make_figure(title='Stream graph'):
    apply_theme(fig_size=(8, 4))
    rng = np.random.default_rng(3)
    n, n_series = 200, 5
    x = np.linspace(0, 24, n)
    Y = np.zeros((n_series, n))
    for i in range(n_series):
        for _ in range(4):                       # 每条河流 = 几个高斯鼓包之和
            c, w, a = rng.uniform(2, 22), rng.uniform(1.5, 5), rng.uniform(0.5, 2)
            Y[i] += a * np.exp(-0.5 * ((x - c) / w)**2)
    base = -Y.sum(axis=0) / 2                    # wiggle 基线（对称轮廓）
    layers = base + np.vstack([np.zeros(n), np.cumsum(Y, axis=0)])
    fig, ax = plt.subplots()
    for i in range(n_series):
        ax.fill_between(x, layers[i], layers[i + 1], color=cycle(i),
                        alpha=0.85, linewidth=0.5, edgecolor='white',
                        label=f'topic {i+1}')
    ax.set_xlabel('time (month)'); ax.set_ylabel('flow magnitude')
    ax.set_title(title)
    ax.legend(frameon=False, ncol=5, loc='upper center',
              bbox_to_anchor=(0.5, -0.15))
    ax.grid(True, axis='x', linestyle=':', alpha=0.5)
    fig.tight_layout()
    return fig

if __name__ == '__main__':
    fig = make_figure(); fig.savefig(__file__.replace('.py', '.png'), dpi=150)
